### PR TITLE
[New Rule] Potential curl CVE-2023-38545 Exploitation

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
Detects potential curl CVE-2023-38545 exploitation by monitoring for vulnerable command line arguments in conjunction with an unusual command line length. A flaw in curl version <= 8.3 makes curl vulnerable to overflow a heap based buffer in the SOCKS5 proxy handshake. Upgrade to curl version >= 8.4 to patch this vulnerability. This exploit can be executed with and without the use of environment variables. For increased visibility, enable the collection of http_proxy, HTTPS_PROXY and ALL_PROXY environment variables based on the instructions within the setup guide of this rule. 

## Detection
```
process where host.os.type == "linux" and event.action == "exec" and event.type == "start" and process.name == "curl" 
and (
  process.args : ("--socks5-hostname", "--proxy", "--preproxy", "socks5*") or 
  process.env_vars: ("http_proxy=socks5h://*", "HTTPS_PROXY=socks5h://*", "ALL_PROXY=socks5h://*")
) and length(process.command_line) > 255
```

image

image

### CODE REVIEW

**Review by terrancedejesus — APPROVED** [review]
Nice turnaround!

File: rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+[metadata]
+creation_date = "2020/10/11"
+integration = ["endpoint"]
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
```
Comment (Aegrah) [inline rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml:6]:
```suggestion
min_stack_comments = "Linux environment variable capture feature via the Elastic Defend Integration was added in 8.6."
min_stack_version = "8.6.0"
```

File: rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml [hunk 2]
```diff
@@ -0,0 +1,78 @@
+[metadata]
+creation_date = "2020/10/11"
```
Comment (w0rk3r) [inline rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml:2]:
```suggestion
creation_date = "2023/10/11"
```

File: rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+with and without the use of environment variables. For increased visibility, enable the collection of http_proxy, 
+HTTPS_PROXY and ALL_PROXY environment variables based on the instructions within the setup guide of this rule.  
+"""
+
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential curl CVE-2023-38545 Exploitation"
+note = """## Setup
```
Comment (w0rk3r) [inline rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml:24]:
```suggestion
setup = """
```

File: rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+Elastic Defend integration does not collect environment variable logging by default.
+In order to capture this behavior, this rule requires a specific configuration option set within the advanced settings
+of the Elastic Defend integration.
```
Comment (w0rk3r) [inline rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml:27]:
```suggestion
In order to capture this behavior, this rule requires a specific configuration option set within the advanced settings of the Elastic Defend integration.
```

File: rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+To set up environment variable capture for an Elastic Agent policy:
+- Go to Security → Manage → Policies.
+- Select an Elastic Agent policy.
+- Click Show advanced settings.
+- Scroll down or search for linux.advanced.capture_env_vars.
+- Enter the names of env vars you want to capture, separated by commas.
+- For this rule the linux.advanced.capture_env_vars variable should be set to "http_proxy,HTTPS_PROXY,ALL_PROXY".
+- Click Save.
+After saving the integration change, the Elastic Agents running this policy will be updated and
+the rule will function properly.
```
Comment (w0rk3r) [inline rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml:37]:
```suggestion
After saving the integration change, the Elastic Agents running this policy will be updated and the rule will function properly.
```

File: rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+For more information on capturing environment variables refer - https://www.elastic.co/guide/en/security/current/environment-variable-capture.html
```
Comment (w0rk3r) [inline rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml:38]:
```suggestion
For more information on capturing environment variables refer to https://www.elastic.co/guide/en/security/current/environment-variable-capture.html
```

**Review by w0rk3r — APPROVED** [review]
🧙🏼

File: rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+maturity = "production"
+min_stack_comments = "Linux environment variable capture feature via the Elastic Defend Integration was added in 8.6."
+min_stack_version = "8.6.0"
+updated_date = "2023/10/11"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects potential curl CVE-2023-38545 exploitation by monitoring for vulnerable command line arguments in conjunction 
+with an unusual command line length. A flaw in curl version <= 8.3 makes curl vulnerable to overflow a heap based buffer
```
Comment (imays11) [inline rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml:13]:
```suggestion
with an unusual command line length. A flaw in curl version <= 8.3 makes curl vulnerable to a heap based buffer overflow
```

File: rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+in the SOCKS5 proxy handshake. Upgrade to curl version >= 8.4 to patch this vulnerability. This exploit can be executed
```
Comment (imays11) [inline rules/linux/execution_curl_cve_2023_38545_heap_overflow.toml:14]:
```suggestion
during the SOCKS5 proxy handshake. Upgrade to curl version >= 8.4 to patch this vulnerability. This exploit can be executed 
```

**Review by imays11 — APPROVED** [review]
Just made a few minor suggestions for readability

**Pull request merged**